### PR TITLE
[Compute] `az disk create`: `--source` add support for creating a disk from disk restore point

### PR DESCRIPTION
**Pull request by ReaNAiveD**
**Related command**

- `az disk create`

**Description**

Feature Request #22850

**Testing Guide**

azdev test test_vm_disk_from_restore_point

**History Notes**

[Component Name 1] BREAKING CHANGE: `az command a`: Make some customer-facing breaking change
[Component Name 2] `az command b`: Add some customer-facing feature

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

File: src/azure-cli/azure/cli/command_modules/vm/_validators.py [hunk 1]
```diff
@@ -1713,14 +1713,31 @@ def validate_vmss_disk(cmd, namespace):
...
+    validate_edge_zone(cmd, namespace)
+    if namespace.source:
+        usage_error = 'usage error: --source {SNAPSHOT | DISK | RESTOREPOINT} | ' \
+                      '--source VHD_BLOB_URI [--source-storage-account-id ID]'
+        try:
+            namespace.source_blob_uri, namespace.source_disk, namespace.source_snapshot, \
+                namespace.source_restore_point, _ = _figure_out_storage_source(
+                    cmd.cli_ctx, namespace.resource_group_name, namespace.source)
+            if not namespace.source_blob_uri and namespace.source_storage_account_id:
+                raise CLIError(usage_error)
```
Comment (wangzelin007) [inline src/azure-cli/azure/cli/command_modules/vm/_validators.py:1728]: Could you please use a specific error type instead of CLIError?

Comment (ReaNAiveD) [reply src/azure-cli/azure/cli/command_modules/vm/_validators.py:1728]: In fact, I split `process_disk_or_snapshot_create_namespace` into `process_disk_create_namespace` and `process_snapshot_create_namespace` due to `--source` should support DiskRestorePoint Id only when creating a disk. `CLIError` was formerly used in `process_disk_or_snapshot_create_namespace`. 
I will replace `CLIError` with `ArgumentUsageError`.

File: src/azure-cli/azure/cli/command_modules/vm/tests/latest/test_vm_commands.py [hunk 2]
```diff
@@ -886,6 +886,48 @@ def test_vm_managed_disk(self, resource_group):
...
+        # create a restore point
+        point = self.cmd('restore-point create -g {rg} -n {point_name} --collection-name {collection_name}').get_output_in_json()
+        self.kwargs.update({
+            'point_id': point['id']
+        })
+        disk_restore_point_id = self.cmd('restore-point show --resource-group {rg} --collection-name {collection_name} --name {point_name} --query \"sourceMetadata.storageProfile.dataDisks[0].diskRestorePoint.id\"').get_output_in_json()
+        self.kwargs['disk_restore_point_id'] = disk_restore_point_id
+        
+        # create disk from restore point
+        self.cmd('disk create --resource-group {rg} --name {disk_name2} --sku {disk_sku} --size-gb {disk_size2} --source {disk_restore_point_id}')
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/vm/tests/latest/test_vm_commands.py:930]: Could you add some check for this command?

Comment (ReaNAiveD) [reply src/azure-cli/azure/cli/command_modules/vm/tests/latest/test_vm_commands.py]: Pr Updated

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**